### PR TITLE
Add Aryn trained DETR model for entity detection

### PR DESCRIPTION
**Pull request by bohou-aryn**

### CODE REVIEW

**Review by baitsguy — COMMENTED** [review]
is it not possible to do table extraction if we use this?

File: sycamore/data/bbox.py [hunk 1]
```diff
@@ -29,3 +30,15 @@ def width(self) -> float:
     @property
     def coordinates(self) -> Tuple[float, float, float, float]:
         return self.x1, self.y1, self.x2, self.y2
+
+    def zero(self) -> bool:
```
[the remark this answers is not in the document]
Comment (bohou-aryn) [reply sycamore/data/bbox.py:34]: not used any more, previously used to check if a bbox is 0, would remove.

File: sycamore/transforms/detr_partitioner.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+
+class SycamorePDFPartitioner:
+    def __init__(self, model_name_or_path):
+        self.pdfminer = PDFMinerExtractor()
+        self.model = DeformableDetr(model_name_or_path)
+
+    @staticmethod
+    def _supplement_text(inferred: List[Element], text: List[Element], threshold: float = 0.5) -> List[Element]:
+        # this is a n^2 time complexity, but for hungarian, it's too expensive? n^3
```
[the remark this answers is not in the document]
Comment (bohou-aryn) [reply sycamore/transforms/detr_partitioner.py]: fixed, similar but simplified, I only supplement text element to the inferred ones.

File: sycamore/transforms/detr_partitioner.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            for extracted_region in text:
+                if ele.bbox and extracted_region.bbox and ele.bbox.iou(extracted_region.bbox) > threshold:
+                    ele.text_representation = extracted_region.text_representation
+                    matched = extracted_region
+            if matched:
+                text.remove(matched)
+        return inferred
+
+    def partition_pdf(self, file: BinaryIO, threshold: float = 0.4) -> List[List["Element"]]:
+        with tempfile.TemporaryDirectory() as tmp_dir, tempfile.NamedTemporaryFile() as tmp_file:
```
[the remark this answers is not in the document]
Comment (bohou-aryn) [reply sycamore/transforms/detr_partitioner.py:39]: I did not make it work directly from bytes, would try couple more times.

**Review by baitsguy — APPROVED** [review]

**Pull request merged**